packages/next/src/build/index.ts

```diff
@@ -2373,33 +2390,27 @@ export default async function build(
                 hasDynamicData = true
                 // we might have determined during prerendering that this page
                 // used dynamic data
-                pageInfos.set(route, {
-                  ...(pageInfos.get(route) as PageInfo),
+                pageInfoRegistry.patch(route, {
```

Comment:
This `.patch` method checks if the underlying page exists first, otherwise it throws. This ensures that we don't put the registry into an invalid state.